script_compiler.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2021 the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+type ScriptCompilerCachedData []byte
+
+type ScriptCompilerCompileOption int
+
+const (
+	ScriptCompilerCompileOptionNoCompileOptions = iota
```

Comment:
This seems quite verbose for something that would ideally just be a default argument that wouldn't need to be specified.

Reply:
This was addressing a comment you made here: https://github.com/Shopify/v8go/pull/13#discussion_r729825955
I agree that it's the default but it is different than `EagerCompile` and I wanted to have our api be consistent with v8 in this area. Specifically, the integer mapping from this enum is 1:1 with the enum in v8.

Reply:
The `ScriptCompiler` prefix is verbose but it would align again with the suggestion you made to call `CachedData` -> `ScriptCompilerCachedData`

Reply:
It looks like this gets referred to as `v8::ScriptCompiler::kNoCompileOptions` when using the V8 API, which avoids the redundant `CompileOption` reference in the name. `ScriptCompilerNoCompileOptions` would actually be closer to how the API gets used in C++.

I do think `CachedData` could be ambiguous.  `ScriptCompilerCachedData` does seem verbose, although at least doesn't introduce redundancy.  Naming it `CompilerCachedData` seems like a better name with the current C++ API, given that function and module compilation also happens on ScriptCompiler.  I'm just not sure why the C++ API didn't name `ScriptCompiler` as `Compiler`.